src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        StringBuilder stringBuilder = new StringBuilder(
+                "Report for period " + dateFrom + " - " + dateTo);
+
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, formatter);
+        LocalDate localDateTo = LocalDate.parse(dateTo, formatter);
+
+        int dataIndex = 0;
+        int nameIndex = 1;
+        int hoursIndex = 2;
+        int salaryPerHourIndex = 3;
+
+        for (String name : names) {
+            int salary = 0;
+            for (String entry : data) {
+                String[] entrySplit = entry.split(" ");
+                LocalDate localOneData = LocalDate.parse(entrySplit[dataIndex], formatter);;
+                String trimmedName = entrySplit[nameIndex];
+
+                if (localOneData.isAfter(localDateFrom) && localOneData.isBefore(localDateTo)
+                        || localOneData.equals(localDateTo) || localOneData.equals(localDateFrom)) {
+
+                    if (trimmedName.equals(name)) {
```

Comment:
There is no trimming, name was extracted from data entry. Just `name` is enough. Remember the `KISS` value - keep names as simple and self-descriptive as possible.